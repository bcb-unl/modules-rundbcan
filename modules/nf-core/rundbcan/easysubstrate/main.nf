process RUNDBCAN_EASYSUBSTRATE {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/dbcan:5.0.4--pyhdfd78af_0' :
        'biocontainers/dbcan:5.0.4--pyhdfd78af_0' }"
    input:
    tuple val(meta), path(input_raw_data)
    tuple val(meta2), path(input_gff), val (gff_type)
    path dbcan_db

    output:
    tuple val(meta), path("${prefix}/overview.tsv"), emit: cazyme_annotation
    tuple val(meta), path("${prefix}/dbCAN_hmm_results.tsv"), emit: dbcanhmm_results
    tuple val(meta), path("${prefix}/dbCANsub_hmm_results.tsv"), emit: dbcansub_results
    tuple val(meta), path("${prefix}/diamond.out"), emit: dbcandiamond_results
    tuple val(meta), path("${prefix}/cgc.gff"), emit: cgc_gff
    tuple val(meta), path("${prefix}/cgc_standard_out.tsv"), emit: cgc_standard_out
    tuple val(meta), path("${prefix}/diamond.out.tc"), emit: diamond_out_tc
    tuple val(meta), path("${prefix}/TF_hmm_results.tsv"), emit: tf_hmm_results
    tuple val(meta), path("${prefix}/STP_hmm_results.tsv"), emit: stp_hmm_results
    tuple val(meta), path("${prefix}/total_cgc_info.tsv"), emit: total_cgc_info
    tuple val(meta), path("${prefix}/CGC.faa"), emit: cgc_faa
    tuple val(meta), path("${prefix}/PUL_blast.out"), emit: pul_blast_out
    tuple val(meta), path("${prefix}/substrate_prediction.tsv"), emit: substrate_prediction
    tuple val(meta), path("${prefix}/synteny_pdf/"), emit: synteny_pdf




    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}_dbcan_substrate"
    def VERSION = '5.0.4'

    """
    mkdir -p ${prefix}
    touch ${prefix}/PUL_blast.out

    run_dbcan easy_substrate \\
        --mode protein \\
        --db_dir ${dbcan_db} \\
        --input_raw_data ${input_raw_data} \\
        --output_dir ${prefix} \\
        --input_gff ${input_gff} \\
        --gff_type ${gff_type} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dbcan: $VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}_dbcan_substrate"
    def VERSION = '5.0.4'
    """
    mkdir -p ${prefix}
    touch ${prefix}/overview.tsv
    touch ${prefix}/dbCAN_hmm_results.tsv
    touch ${prefix}/dbCANsub_hmm_results.tsv
    touch ${prefix}/diamond.out
    touch ${prefix}/cgc.gff
    touch ${prefix}/cgc_standard_out.tsv
    touch ${prefix}/diamond.out.tc
    touch ${prefix}/TF_hmm_results.tsv
    touch ${prefix}/STP_hmm_results.tsv
    touch ${prefix}/total_cgc_info.tsv
    touch ${prefix}/CGC.faa
    touch ${prefix}/PUL_blast.out
    touch ${prefix}/substrate_prediction.tsv
    mkdir -p ${prefix}/synteny_pdf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        dbcan: $VERSION
    END_VERSIONS
    """
}
