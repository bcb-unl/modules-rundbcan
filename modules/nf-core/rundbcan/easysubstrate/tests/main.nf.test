// nf-core modules test dbcan
nextflow_process {

    name "Test Process RUNDBCAN_EASYSUBSTRATE"
    script "../main.nf"
    process "RUNDBCAN_EASYSUBSTRATE"

    tag "modules"
    tag "modules_nfcore"
    tag "rundbcan"
    tag "rundbcan/database"
    tag "rundbcan/easysubstrate"

    test("easycgc - simplified") {

        setup {
            run("RUNDBCAN_DATABASE"){
                script "../../database/main.nf"
                process {
                    """
                    """
                }

        }

    }

        when {
            process {
                """

                input[0] = [
                    [id:'test_dbcan'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/proteome.fasta', checkIfExists: true)
                ]

                input[1] = [
                    [id:'test_dbcan'],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/candidatus_portiera_aleyrodidarum/genome/gff/test1.gff', checkIfExists: true),
                    "prodigal"

                ]

                input[2] = RUNDBCAN_DATABASE.out.dbcan_db
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }

            )
        }
    }

    test("easysubstrate - stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [[id: 'stub'], file('stub')]
                input[1] = [[id: 'stub'], file('stub.gff'), "prodigal"]
                input[2] = file('stub_db')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
            )
        }
    }
}
