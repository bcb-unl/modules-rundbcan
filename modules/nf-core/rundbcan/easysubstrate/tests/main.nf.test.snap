{
    "easycgc - simplified": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_overview.tsv:md5,73bd9acee752d61e096370d4fedfee54"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_dbCAN_hmm_results.tsv:md5,7ce7b6536845ffa8907b3f3fb2b77a1b"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_CGC.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_PUL_blast.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_substrate_prediction.tsv:md5,fe2a5ea9e19c4f1108798103547ff98d"
                    ]
                ],
                "13": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        [
                            
                        ]
                    ]
                ],
                "14": [
                    "versions.yml:md5,dc423c05a1d969f28a53bd6bd1e2c2e6"
                ],
                "2": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_dbCANsub_hmm_results.tsv:md5,7ce7b6536845ffa8907b3f3fb2b77a1b"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_diamond.out:md5,c935cda6778ca2b6aaaa4362b6f24d84"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_cgc.gff:md5,cb1bd08c0276b4a0a37540032863a0ac"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_cgc_standard_out.tsv:md5,6be9ab29b289ff46cc6e4b6fe48dc3d7"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_diamond.out.tc:md5,4b9747475aaf438eede8556832060f83"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_TF_hmm_results.tsv:md5,f63f2b2b3c4439304fa1c20c19cf1e99"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_STP_hmm_results.tsv:md5,e03c881b99bb7c7637ed600324816b5c"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_total_cgc_info.tsv:md5,79377e2e3f9582b3faa5f22996868bfa"
                    ]
                ],
                "cazyme_annotation": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_overview.tsv:md5,73bd9acee752d61e096370d4fedfee54"
                    ]
                ],
                "cgc_faa": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_CGC.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cgc_gff": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_cgc.gff:md5,cb1bd08c0276b4a0a37540032863a0ac"
                    ]
                ],
                "cgc_standard_out": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_cgc_standard_out.tsv:md5,6be9ab29b289ff46cc6e4b6fe48dc3d7"
                    ]
                ],
                "dbcandiamond_results": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_diamond.out:md5,c935cda6778ca2b6aaaa4362b6f24d84"
                    ]
                ],
                "dbcanhmm_results": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_dbCAN_hmm_results.tsv:md5,7ce7b6536845ffa8907b3f3fb2b77a1b"
                    ]
                ],
                "dbcansub_results": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_dbCANsub_hmm_results.tsv:md5,7ce7b6536845ffa8907b3f3fb2b77a1b"
                    ]
                ],
                "diamond_out_tc": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_diamond.out.tc:md5,4b9747475aaf438eede8556832060f83"
                    ]
                ],
                "pul_blast_out": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_PUL_blast.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "stp_hmm_results": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_STP_hmm_results.tsv:md5,e03c881b99bb7c7637ed600324816b5c"
                    ]
                ],
                "substrate_prediction": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_substrate_prediction.tsv:md5,fe2a5ea9e19c4f1108798103547ff98d"
                    ]
                ],
                "synteny_pdf": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        [
                            
                        ]
                    ]
                ],
                "tf_hmm_results": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_TF_hmm_results.tsv:md5,f63f2b2b3c4439304fa1c20c19cf1e99"
                    ]
                ],
                "total_cgc_info": [
                    [
                        {
                            "id": "test_dbcan"
                        },
                        "test_dbcan_dbcan_substrate_total_cgc_info.tsv:md5,79377e2e3f9582b3faa5f22996868bfa"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dc423c05a1d969f28a53bd6bd1e2c2e6"
                ]
            },
            {
                "RUNDBCAN_EASYSUBSTRATE": {
                    "dbcan": "5.0.6"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-29T12:19:09.562405"
    },
    "easysubstrate - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_overview.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_dbCAN_hmm_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_CGC.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "11": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_PUL_blast.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "12": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_substrate_prediction.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "13": [
                    [
                        {
                            "id": "stub"
                        },
                        [
                            
                        ]
                    ]
                ],
                "14": [
                    "versions.yml:md5,dc423c05a1d969f28a53bd6bd1e2c2e6"
                ],
                "2": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_dbCANsub_hmm_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_diamond.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_cgc.gff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_cgc_standard_out.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_diamond.out.tc:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_TF_hmm_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_STP_hmm_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_total_cgc_info.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cazyme_annotation": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_overview.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cgc_faa": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_CGC.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cgc_gff": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_cgc.gff:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "cgc_standard_out": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_cgc_standard_out.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "dbcandiamond_results": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_diamond.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "dbcanhmm_results": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_dbCAN_hmm_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "dbcansub_results": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_dbCANsub_hmm_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "diamond_out_tc": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_diamond.out.tc:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "pul_blast_out": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_PUL_blast.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "stp_hmm_results": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_STP_hmm_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "substrate_prediction": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_substrate_prediction.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "synteny_pdf": [
                    [
                        {
                            "id": "stub"
                        },
                        [
                            
                        ]
                    ]
                ],
                "tf_hmm_results": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_TF_hmm_results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "total_cgc_info": [
                    [
                        {
                            "id": "stub"
                        },
                        "stub_dbcan_substrate_total_cgc_info.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dc423c05a1d969f28a53bd6bd1e2c2e6"
                ]
            },
            {
                "RUNDBCAN_EASYSUBSTRATE": {
                    "dbcan": "5.0.6"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-29T12:19:13.757388"
    }
}